lib/screens/e_commerce/promo_item.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:carousel_slider/carousel_slider.dart';
+
+class PromoItem extends StatelessWidget {
+  const PromoItem({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    Size size = MediaQuery.of(context).size;
+    return Container(
+      // color: lightGrey,
+      height: size.height * 0.26,
+      // width: width * 2,
+      child: ListView(
```

Comment:
These are items (actually products) which have been marked as on promo. So we shalll actually fetch them from the server and display here, so you could adjust this so that you are using data from json, then just loop through the data. So the idea shouldn't be so different from the products. Of course there would be some more detail about the item on promotion.